### ISSUE TITLE
Undefined array key "type" when removing blocks from the page

### ISSUE THREAD

**Issue by webspilka** [Z3d0X/filament-fabricator#144]
hi 
my stacktrace: https://flareapp.io/share/v5pba3lP

I use 
filament                 v3.2.44
livewire/liveire       v3.4.7
spatie/laravel-translatable                 v.6.6.2
z3d0x/filament-fabricator                   v2.1.1
filament/spatie-laravel-translatable-plugin v3.2.44

I was so happy when I connected everything and the translation worked.  But after a while I found the problem.
Unfortunately, removing blocks from the page does not work

When I delete block and push save, I get error

```
Undefined array key "type"
```

<img width="1082" alt="Screenshot 2024-03-07 at 13 27 47" src="https://github.com/Z3d0X/filament-fabricator/assets/1138069/9bef109e-dab0-45a0-8e5f-e6740b050f38">

**Comment by webspilka** [comment]
i find problem in saved data in DB
I have Pages.blocks 
```json
{
  "uk":{
    "625c288b-b90c-410b-a292-bf7e3d045669":{
      "type":"my",
      "data":{
        "name":"test3 page",
        "content":"<p>test 3 page</p>"
      }
    },
    "1287e0f6-4ee9-4831-a7bc-7c9ede585e16":{
      "type":"my",
      "data":{
        "name":"test 3 block2",
        "content":"<p>test 3 block2</p>"
      }
    }
  },
  "en":{
    "1287e0f6-4ee9-4831-a7bc-7c9ede585e16":{
      "data":{
        "content":""
      }
    },
    "625c288b-b90c-410b-a292-bf7e3d045669":{
      "data":{
        "content":""
      }
    }
  },
  "ru":[
    {
      "type":"my",
      "data":{
        "name":"ru-block1",
        "content":"<p>ru-block1</p>"
      }
    }
  ]
}
```
when i manually remove english localization everything starts working. But then again after creating and deleting several blocks, the structure becomes invalid

**Comment by webspilka** [comment]
At first, I thought the issue arose due to the plugin `"filament/spatie-laravel-translatable-plugin": "^3.2",` being connected. However, upon disconnecting thtis plugin, I encountered the same problem.

I began testing different scenarios and found that everything hinges on the `Filament\Forms\Components\RichEditor`. If the RichEditor field contains text, then upon deleting the block and saving the page, I receive an error. If I first clear the RichEditor of text and then delete the block, everything works without issues.

https://github.com/Z3d0X/filament-fabricator/assets/1138069/e0527034-bb33-489c-a8ba-88d27ffa7e59

I see that a similar issue was present before:  https://github.com/filamentphp/filament/issues/7957
